### ISSUE TITLE
find missing egauge data

### ISSUE THREAD

**Issue by eileenpeppard** [erdl/legacy-scrape-util#12]
We have no data from eGauge 792 (installed 8/12), but I'm sure Jason and I added that one to the toml file. I'm attaching what text I believe we added.

Also, I really need data from other egauges (795, 785, 786) from before 2017-09-19 16:18.  I installed these on 8/12, so missing more than a month of data. 

toml file text:
```
'e785' = 785, 'e786' = 786, 'e795' = 795, 'e792' = 792

'e785-Whole-house' = 187
'e785-Water-heater' = 188
e785-ACC = 189
e785-AHU = 190
e785-Dryer = 191
e785-Range = 192
'e785-Kitchen-refrigerator' = 193
e785-Other = 194

e786-PV	= 195

'e795-Whole-house' = 196
'e795-Water-heater' = 197
e795-ACC = 198
e795-AHU = 199
e795-Dryer = 200
e795-Range = 201
'e795-Kitchen-refrigerator' = 202
e795-Other = 203

e792-PV	= 204
```

**Comment by carlosviansi** [comment]
@jygh98 I thought after we confirmed #11 you used the code on #6 which contained those 12 python files from scrape-utils to insert the data and told everything was fine. How come they are still missing?

**Comment by jygh98** [comment]
Okay I sent you the data on uh file drop, so this is the formatted data from 8/12 to 9/19 for egauges 795, 785, 786.  Currently we are not able to get any data from e792 despite it being up so I will have to look into that.

**Comment by eileenpeppard** [comment]
Thanks for the data, Jason. I don't know why e792 can't be accessed. The eGuard says its online 77% of the time... i'll check it periodically, but the cron job should have found it online by now.

**Comment by eileenpeppard** [comment]
I just called eGauge support and left a message. EGuard says its online right now and i can't access it.

**Comment by carlosviansi** [comment]
@eileenpeppard I checked scrape-utils timestamps yesterday night with @jygh98 of the last time it had access to it was at unix timstamp `1505873821.2361722`, which is equivalent to `09/20/2017 @ 2:17am (UTC)`. 09/20 is the date you and Jason added `e792` to the egauge list.

I am led to believe it tried running that time, set the timestamp as part of it's the first attempt in spite of not downloading any data, and then never again despite multiple attempts to talk to egauge`s API.

It doesn't seem like a global API problem, just with e792:

```
e34105 = 1507083360.0
e786 = 1507097760.0
e34104 = 1507086900.0
e785 = 1507097760.0
e34108 = 1507097760.0
e34110 = 1507097760.0
e34106 = 1507097760.0
e34111 = 1507097760.0
e790 = 1507097760.0
e792 = 1505873821.2361722
e34109 = 1507097760.0
e34113 = 1507097760.0
e34107 = 1507097760.0
e795 = 1507097760.0
``` 

And you are right, if it was at least 2 consecutive hours available, scrape-util should have been able to add more data since 09/20:

<img width="624" alt="screenshot 2017-10-03 20 42 56" src="https://user-images.githubusercontent.com/17270563/31204085-c85ac854-a905-11e7-8485-7443e7f98627.png">

It would then either boil down to how you and Jason added the sensors, or their availability report system is broken in the sense "available" doesn't mean "data can be collected", as there are up to 3 consecutive hours of availability every day. 

Given that the other sensors work, and you just mentioned even on available you can't manually download data, I am led to believe the problem is on their end.

**Comment by eileenpeppard** [comment]
![egauge permissions issue](https://user-images.githubusercontent.com/29083044/31210259-9b864ffc-a92c-11e7-9ada-e560f1bc3f08.PNG)

**Comment by carlosviansi** [comment]
Alright, lads, the problem is solved and we now have all possible data we could. I met with Eileen from 4 something PM to 6 PM and here is the memory of it in case of incidence. Breaking in parts:

## eGauge792 has no missing data

> We have no data from eGauge 792 (installed 8/12), but I'm sure Jason and I added that one to the toml file. 

A couple of evidence led for us to figure this out (and I am letting you know so you keep an eye for this next time):

1. First, I checked the egauge state on /var/local (the file that has the most recent unix timestamp for each egauge and shown in the post above). e792 was strangely way back in time and was different than the others. 
1. I tried to request data via the erdl/sensors/egauge notebook, and rather than a timeout I was getting no authorization. Via browser that also didn't work. 
1. Eileen noticed to change this egauge configuration she didn't require password. 

After she looked around, she found this box she highlights was `checked`, while **all the other sensors were NOT**. To get to this page, you would need the login/password for the eGauge dashboard, which only Eileen and I have. Then:
   1. You would click a given egauge to see it's live graph.
   1. Then on the live graph page to the right upper corner, there is a Settings button. Clicking takes to a screen with the same left menu.
   1. Then you click `Access Control` on the left menu to be taken to this screen.  

**Solution**: Unchecking the box removed the authentication requirement. Note Eileen didn't know there was a request for authentication to get the data because her browser had some master/username password already stored that wouldn't prompt the pop-up screen (which of course scrape-util couldn't answer).

I ran my Notebook and now I had access, so scrape-util should function accordingly. 

### e792 eGauge config.toml variable names.

I asked Jason where did he get the new sensor names from, and he said he just followed the template of the other egauge sensor names. However, there is a place where the sensor name does exist, which is also on this online interface. You need to remember eGauge does not magically provide sensor of appliances (e.g. AC, dryer, PV, etc) but has 12 CTs (Current Transducers) that are hooked on cables that feed energy to this appliance.

It is **Eileen**, that through the `Settings` page of the website, provide the equations on how to sum the different `CTs` and **also name them as appliances**. As such, when you are typing in `config.toml` the suffix such as `-PV`, you are really referring to the name Eileen`s name in the configuration she set up for a given Sensor. 

TL;DR: Every time a sensor is added to the config.toml file, it should be part of the process (which needs to be documented in the wiki eventually, I am opening an issue for this), check whatever was in the configuration of the file to make sure the name **matches**. Do `not` go by convention! 

### e792 will be missing **correct** data up to today (October 4, 2017) at 5 PM. 

While troubleshooting e792, which to my understanding is dedicated solely to PV, Eileen realized her equations for the CTs were incorrect. Originally it's 2 CTs were supposed, to sum up in what would be our PV Power (kW). However, the equation was of subtraction between the CTs, thus canceling them out leading to values close to `0`. As such, anything that looks like to 0 is `incorrect` from any point of.
There is currently no conventional way to flag this in the database such as a `quality check column`. Maybe this can be considered down the road. 

##  egauges (795, 785, 786) from before 2017-09-19 16:18 missing data.

Also, I really need data from other egauges (795, 785, 786) from before 2017-09-19 16:18. I installed these on 8/12, so missing more than a month of data.

There was miscommunication between Jason and Eileen on the time range when Jason was first asked about the data. As such, Eileen only received data of one day, rather than all the time range before. This was addressed yesterday night after I talked to Jason. 

Noteworthy here, our current pipeline to get data from before is still as clunky as it can get because I didn't have time to finish the notebooks with this trip:

1. Eileen specifies a time range to Jason.
1. Jason uses the 12 python notebook with Scrape-Util to get some data in the readings table format.
1. Jason then e-mail back the data to Eileen.
1. Eileen uploads the data to the database by hand.

Once I finish the notebook, this should be as simple as:

1. Eileen types time range on Notebook and database name.
1. Notebooks downloads raws data and store in the specified database.

Until this doesn't happen, we will have to stick with the clunky way. Hopefully when I return I can work on this on the following weekend (not this one). 

I am closing the issue since the main problem (missing data) is solved.
